nodejs/koa/routes.js

```diff
@@ -0,0 +1,100 @@
+const Router     = require('koa-router');
+const router     = new Router();
+const passport   = require('koa-passport');
+const connection = require('./db');
+
+let pool;
+
+async function generateBooksTable() {
+
+  let tableHTML = '<table><tr><th>Title</th><th>Author</th></tr>';
+
+  // get the books from the database
+  const results = await pool.query("SELECT * FROM KOA_BOOKS");
```

Comment:
If we're having the user log in with IBM i username/password, we should use those same credentials for the database access. 

As written, the security model of this app is inherently broken, because:
- It requires the server user to have the appropriate authority
- It simply verifies that the user has any valid userid/password. There could be user profiles that should not have access to data

If this app were to follow best practices:
- The server user should only have enough authority that is required for the application to run. In fact, the server user shouldn't be able to access any business data
- Data access should be governed on a per-user basis (essentially getting those elevated privileges through the login)

A better design would be to use the username/password to get a connection, then use that connection for the user's session.

Reply:
Good points! I pushed some changes before I saw this, but let me do a little tweaking. This will be nice as well because I can remove `itoolkit` as a dependency

Reply:
Changed in latest changes